src/register.js

```diff
@@ -0,0 +1,56 @@
+const { v4 } = require("uuid");
+const AWS = require("aws-sdk");
+const { response, isValidEmail } = require("./utils");
+const bcrypt = require("bcryptjs");
+
+const dynamo = new AWS.DynamoDB.DocumentClient();
+
+const register = async (event) => {
+  const userId = v4();
+  try {
+    const { email, password } = JSON.parse(event.body);
+    if (!email || !password) {
+      return response(401, { message: "Email or password is empty" });
+    }
+
+    if (!isValidEmail(email)) {
+      return response(401, { message: "Invalid email" });
+    }
+
+    const params = {
+      TableName: "TodoTable",
+    };
+
+    const scanResults = [];
+    const items = await dynamo.scan(params).promise();
+    items.Items.forEach((item) => scanResults.push(item));
+    params.ExclusiveStartKey = items.LastEvaluatedKey;
+
+    for (const results in scanResults) {
```

Comment:
This is an option you have and is good but there is another that is most efficient. You can do this validation when you query dynamo DB using the FiltersExpression field. Take a look and try to implement also the other solution I suggested